### PR TITLE
Better printing of empty signatures

### PR DESCRIPTION
**Pull request by gasche**
This is the unobjectionable, easy-to-review part of my tentative to give more structure to the code printing functor types -- see #8906.

### CODE REVIEW

File: testsuite/tests/typing-extensions/msg.ocaml.reference [hunk 1]
```diff
@@ -12,7 +12,7 @@ module Msg :
         val write : t -> string
         val read : string -> t
       end
-    module Define : functor (D : Desc) -> sig type 'a tag += C : D.t tag end
+    module Define : functor (D : Desc) sig type 'a tag += C : D.t tag end
```
Comment (Octachron) [inline testsuite/tests/typing-extensions/msg.ocaml.reference]: This arrow should not be removed.

File: testsuite/tests/tool-ocamldoc/Inline_records.man.reference [hunk 2]
```diff
@@ -1,3 +1,4 @@
+.TH "Inline_records" 3 2019-09-02 OCamldoc ""
```
Comment (Octachron) [inline testsuite/tests/tool-ocamldoc/Inline_records.man.reference]: The CI failure is probably due to this line, `ocamltest -promote` is not aware that the ocamldoc's output comparison skips the first line of the output.

**Comment by gasche** [comment]
> The CI failure is probably due to this line, `ocamltest -promote` is not aware that the ocamldoc's output comparison skips the first line of the output.

Indeed, I fixed this is #8913.

**Comment by Octachron** [comment]
Concerning the first renaming commit, I have been reading  `print_simple_out_type` as `print a simple Outcometree.out_type` and not as `simply output a type`. The same reading also holds for `print_simple_out_module_type` :  `print a simple Outcometree.out_module_type`.

The naming convention might be confusing but it seems quite internally consistent, so I feel that at least the commit message should be changed.

**Comment by gasche** [comment]
@Octachron that's fair; I removed the first commit from the PR.

**Review by Octachron — APPROVED** [review]
Rendering empty signatures as `sig␣end` rather than `sig␣␣end` seems more natural (and it is also the rendering chosen by ocamlformat).

**Comment by gasche** [comment]
Thanks! I merged -- a Changes entry is missing, but the CI seems a bit loaded today so I'll just add one for the follow-up #8911.

**Pull request merged**